### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by kolya2731**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,15 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> Any:
```
Comment (Dimosphen1) [inline app/main.py:9]: You could use a decorator from `functools` to preserve the wrapped function name and docstring. Please, use it here

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        if args not in cache_dict:
```
Comment (viktoria-rybenchuk) [inline app/main.py]: How do you check if the given argument is a keyword argument and whether it is contained within a cache?

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]

**Review by pavlopro — APPROVED** [review]
Good Job!

**Pull request closed without merging**